sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -242,6 +245,32 @@ public async ValueTask WaitForEnvironmentAsync()
                 }
                 await task;
             }
+
+            if (!GlobalIsRunningInCI &&
+                Mode is RecordedTestMode.Live or RecordedTestMode.Record &&
+                !s_initializedEnvironmentTypes.Contains(_type))
+            {
+                s_initializedEnvironmentTypes.Add(_type);
+
+                string resourceGroup = GetOptionalVariable("RESOURCE_GROUP");
+                if (resourceGroup == null)
+                {
+                    return;
+                }
+
+                ArmClient armClient = new ArmClient(Credential);
+                ResourceGroup rg = await armClient.DefaultSubscription.GetResourceGroups().GetAsync(resourceGroup);
```

Comment:
Unpopular opinion but should we avoid taking the dependency on ARM client library and have a very simplified client of our own based on Azure.Core pipeline? I'm not sure the test framework having a dep on libraries being tested is a good thing.

Reply:
I'm not sure how feasible the alternative is, or why taking a dependency on ARM client is even a problem.

Reply:
The rest calls in question are really simple I did it for metrics client: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/monitor/Azure.Monitor.Query/tests/MetricsSenderClient.cs

Reply:
We already had issues with perf tooling trying to varry the Azure.Core version and us being tied to latest and being broken by that. A dependency on ARM client might create a similar thing where it get's harder to make changes in ARM client because now the test framework depends on it.

Reply:
We are referencing a specific version of the ARM client. I don't see why this should impact the ARM client evolution. We can just leave the version as-is until there is a GA version that we want to migrate to.

Reply:
Because ARM library tests would use this test framework.

Reply:
Ok I see your point. @m-nash what do you think here?

Reply:
I removed the dependency on ResourceManager.